src/history/hs.c

```diff
@@ -1176,17 +1173,19 @@ __wt_hs_delete_key(WT_SESSION_IMPL *session, uint32_t btree_id, const WT_ITEM *k
         return (0);
 
     WT_RET(__wt_hs_cursor(session, &session_flags, &is_owner));
+
     /*
      * In order to delete a key range, we need to be able to inspect all history store records
      * regardless of their stop time pairs.
      */
-    F_SET(session, WT_SESSION_IGNORE_HS_TOMBSTONE);
+    F_SET(session->hs_cursor, WT_CURSTD_IGNORE_TOMBSTONE);
+
     /* The tree structure can change while we try to insert the mod list, retry if that happens. */
     while ((ret = __hs_delete_key_int(session, btree_id, key)) == WT_RESTART)
         ;
 
-    if (!FLD_ISSET(session_flags, WT_SESSION_IGNORE_HS_TOMBSTONE))
-        F_CLR(session, WT_SESSION_IGNORE_HS_TOMBSTONE);
+    F_CLR(session->hs_cursor, WT_CURSTD_IGNORE_TOMBSTONE);
```

Comment:
Did you choose a cursor flag instead of a session flag because we've run out of session flags?

Reply:
That's part of the reason. Actually we now have an extra flag in the session after WT-5977 get merged. However, I feel we can get more control if we make it a cursor flag. We can ignore tombstones on tables which we really want to do that instead of just ignoring tombstones for every table.